rep-0151.rst

```diff
@@ -0,0 +1,260 @@
+REP: 151
+Title: Python 3 Support
+Author: Dirk Thomas, Shane Loretz
+Status: Draft
+Type: Standards Track
+Content-Type: text/x-rst
+Created: 05-Aug-2019
+
+Outline
+=======
+
+#. Abstract_
+#. Motivation_
+#. Context_
+#. Proposal_
+#. References_
+#. Copyright_
+
+Abstract
+========
+
+This REP describes the path for switching from Python 2 to Python 3 in ROS 1.
+
+Motivation
+==========
+
+As described in the Python release schedule [1]_ Python 2.7 will only receive
+bugfix releases until 2020.
+As a consequence Ubuntu is trying to demote Python 2.7 into the `universe`
+repository [2]_.
+While it is unlikely that Python 2.7 will be removed from the archives entirely,
+it should not be relied on past its EOL date.
+Therefore ROS 1 must move to Python 3.
+
+Context
+=======
+
+Different versions of Python can't be mixed within a single ROS distribution.
```

Comment:
Can we please remove this sentence?  It runs counter an extremely broad body of evidence that single codebases can support both Python 2 and Python 3.

Reply:
That is not what the sentence tries to express.

You can't install a package `foo` with Python 2 and a package `bar` with Python 3 and expect that either of them will be able to find the modules of the other package.

Reply:
"_Python 2 and Python 3 maintain separate _site-package_ directories so packages installed with pip2 can not be found with Python 3 and vice versa._"

Reply:
Exactly, see the other comment why that is a problem: https://github.com/ros-infrastructure/rep/pull/201#discussion_r311735555

Reply:
I deleted this paragraph in d48612e6414428242ae4cb2847247343ad9b48ee